packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -454,7 +454,8 @@ export class AskarWallet implements Wallet {
 
           const keyPublicBytes = key.publicBytes
           // Store key
-          await this.session.insertKey({ key, name: TypedArrayEncoder.toBase58(keyPublicBytes) })
+          const id = kid ?? TypedArrayEncoder.toBase58(keyPublicBytes)
```

Comment:
Ideally we don't store keys by kid as that means we have different formats to store keys (which will get messy).

Maybe we can add the kid as metadata and tag to the key?

Otherwise we won't be able to find some keys by it's public key anymore, because we use a kid

Reply:
I think this can be reverted once we get `DID Resolver` available inside of unpack function.
I added an ability to store keys by `kid` in order to write tests against test vectors from the specification.
Test data there uses `did:example:alice#key-p521-1` key references.

Reply:
We can store it as a tag on the key right? 

Just want to make sure we're not changing the way ids are created / managed (we should probably change the publicKeyBase58 to be a tag as well)

But even then, if we create a key, then later add it to a did document, it wouldn't necessarily have a kid. 

Maybe we need to add something to the did documents we have stored locally so we can do the following:
- query DidRecord by kid
- extract public key from did document
- query for key

I think we should avoid using the did resolver in the wallet itself

Reply:
Yeah. we can find a way to avoid using DidResolver for getting `recipient key` but we cannot for `sender key` as it is unknown (at least when we receive the first message). 

The current assumption is that the sender uses either `did:key` or `did:peer` methods. So the public key can be resolved from the `skid` fingerprint.  
But if some other method is used we need to resolve the DidDocument for the `skid`.`

Reply:
As for `createKey` method changes: I can revert the change and mock values in the tests (similarly to the senderKey).  

`did:peer` method is used in the updated Alice/Faber demo workflow (sending ping message).  Keys (both ED25519 and X25519) are stored by their base58 representation.

Reply:
> skid is inside of base64 encoded protected JWE field. In my opinion, it is not a good idea to parse JWE -> fetch skid -> resolve sender DidDoc and key before calling wallet.unpack function.
Currently, JWE is constructed and parsed in wallet.pack/upack methods.

I think that is fine to do. We should really extract JWE out of the wallet in the coming months (we've been adding a `crypto/jose` directory with `jwk`, `jwt`, `jwa`, `jws`. Adding JWE is the missing piece.

For now I think we should extract the kid on a higher layer and pass it in. OR something I've recently added to the jws code is a `jwkResolver`, which you can optionally pass and will be called when the key used to sign a message can't be determined (the JWS service only supports if the `jwk` properyt is used in the header, for all other purposes you need to pass a `jwkResolver`).

Maybe something like that would work for now. Then when we extract it, we could use the same approach, but as a separate JWE service along side the JWS service